onnxmltools/convert/common/_topology.py

```diff
@@ -726,26 +726,35 @@ def convert_topology(topology, model_name, doc_string, target_opset, targeted_on
             # Convert the selected operator into some ONNX objects and save them into the container
             _registration.get_converter(operator.type)(scope, operator, container)
 
-    # When calling ModelComponentContainer's add_initializer(...), nothing is added into the input list. However, in
-    # ONNX initializers should also be model's (GraphProto) inputs. Thus, we create ValueInfoProto objects from
-    # initializers (type: TensorProto) directly and then add them into model's input list.
-    extra_inputs = []  # ValueInfoProto list of the initializers
-    for tensor in container.initializers:
-        # Sometimes (especially when creating optional input values such as RNN's initial hidden state), an initializer
-        # is also one of the original model's input, so it has been added into the container's input list. If this is
-        # the case, we need to skip one iteration to avoid duplicated inputs.
-        if tensor.name in [value_info.name for value_info in container.inputs]:
-            continue
-
-        # Initializers are always tensors so we can just call make_tensor_value_info(...)
-        value_info = helper.make_tensor_value_info(tensor.name, tensor.data_type, tensor.dims)
-        extra_inputs.append(value_info)
+    # When calling ModelComponentContainer's add_initializer(...), nothing is added into the input list.
+    # However, for ONNX target opset < 9, initializers should also be model's (GraphProto) inputs.
+    # Thus, we create ValueInfoProto objects from initializers (type: TensorProto) directly and
+    # then add them into model's input list.
+    if container.target_opset < 9:
```

Comment:
why not follow the changed was made on keras converter? This change actually breaks the optimize_onnx because it needs to know the initializer list.

Reply:
Just realized after talking to David! Pushed an update with the fix.